Alignment/APEEstimation/plugins/ApeTreeCreateDefault.cc

```diff
@@ -0,0 +1,473 @@
+
+// -*- C++ -*-
+//
+// Package:    DefaultApeTree
+// Class:      DefaultApeTree
+// 
+/**\class ApeTreeCreateDefault ApeTreeCreateDefault.cc Alignment/APEEstimation/src/ApeTreeCreateDefault.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Marius Teroerde (code from ApeEstimator.cc and
+//                   ApeEstimatorSummary.cc)
+//         Created:  Tue Nov 14 11:43 CET 2017
+//
+//
+
+
+// system include files
+#include <memory>
+#include <fstream>
+#include <sstream>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+
+#include "CLHEP/Matrix/SymMatrix.h"
+
+#include "CondFormats/AlignmentRecord/interface/TrackerAlignmentErrorExtendedRcd.h"
+#include "CondFormats/Alignment/interface/AlignmentErrorsExtended.h"
+
+//...............
+#include "Alignment/APEEstimation/interface/TrackerSectorStruct.h"
+#include "Alignment/APEEstimation/interface/ReducedTrackerTreeVariables.h"
+
+#include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+#include "DataFormats/SiStripDetId/interface/SiStripDetId.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/SiStripCluster/interface/SiStripCluster.h"
+
+
+#include "TString.h"
+#include "TFile.h"
+#include "TDirectory.h"
+#include "TTree.h"
+#include "TMath.h"
+//
+// class declaration
+//
+
+class ApeTreeCreateDefault : public edm::one::EDAnalyzer<> {
+  public:
+    explicit ApeTreeCreateDefault(const edm::ParameterSet&);
+    ~ApeTreeCreateDefault() override;
+  
+  
+  private:
+    void beginJob() override ;
+    void analyze(const edm::Event&, const edm::EventSetup&) override;
+    void endJob() override ;
+    
+    void sectorBuilder();
+    bool checkIntervalsForSectors(const unsigned int sectorCounter, const std::vector<double>&)const;
+    bool checkModuleIds(const unsigned int, const std::vector<unsigned int>&)const;
+    bool checkModuleBools(const bool, const std::vector<unsigned int>&)const;
+    bool checkModuleDirections(const int, const std::vector<int>&)const;
+    bool checkModulePositions(const float, const std::vector<double>&)const;
+    
+    // ----------member data ---------------------------
+    const edm::ParameterSet parameterSet_;
```

Comment:
I think, it is considered bad practice to store the full parameter set, as they are quite heavy.
The better approach would be to store only what you need in the initializer list of the constructor and then drop the parameter set.